### ISSUE TITLE
Support Blazor properly for All ASP.Net Identity Scaffolded pages (i.e., not Razor Pages)

### ISSUE THREAD

**Issue by Webreaper** [dotnet/aspnetcore#35551]
I am trying to implement ASP.Net Identity account management etc within my [Blazor Server application](http://github.com/webreaper/damselfly). However, due to the fact that currently ASP.Net identity only supports razor pages (i.e., cshtml), it's **really** difficult to make a seamless experience from a UI perspective. 

For example, I'd like to use [MudBlazor](http://mudblazor.com) controls on my Login and Register pages, with all their UI loveliness - but that's impossible. It means that the user is subjected to a jarring UI paradigm shift when they go from the login page to the rest of the app. 

It also breaks the Blazor development model and means I have to delve into the world of page redirects and callbacks etc., to make stuff work.  The whole point about Blazor is that I don't want to have to deal with all that stuff, but it's unavoidable. 

I've read through the discussions in [13601](https://github.com/dotnet/aspnetcore/issues/13601) and [34095](https://github.com/dotnet/aspnetcore/issues/34095), and I understand some of the arguments being made, but it feels to me like MSFT isn't really embracing the Blazor strategy properly if it doesn't provide a 'native Blazor' way to use Identity.

Totally understand this is something that would take a major amount of work, and so might not be a thing until .Net 7 or even 8, but it feels like it should be something in the pipeline. As more complex UI paradigms become simpler to implement with Blazor, it's a bit sad that login, register, user-management and other areas end up being second-rate citizens of the new world.

**Comment by jooslouw** [comment]
I agree with @Webreaper, I spent days implementing a Blazor (WASM) solution to have a seamless Blazor-only UI experience to my project but still using Identity Server back-end and security to my back-end API.

It doesn't make sense to have all the Blazor goodness of `<AuthorizeView>` components but breaking the UI with redirects to server-side MVC.

**Comment by pranavkm** [comment]
FYI @danroth27

**Comment by nbiada** [comment]
In the meantime you can _mix_ the old cshtml with razor components.  
I use this approach to maintain the same _graphic layout_ between the two systems.

An example, the following file is ___Layout.cshtml__ used by _Identity_.  I
I've used various Blazor components via static rendering:

```cshtml
@using Microsoft.AspNetCore.Hosting
@using Microsoft.AspNetCore.Mvc.ViewEngines
@inject IWebHostEnvironment Environment
@inject ICompositeViewEngine Engine
@addTagHelper *, Microsoft.AspNetCore.Mvc.TagHelpers
@using Project.Server.Shared

<!DOCTYPE html>
<html>
<head>
    <component type="typeof(MainLayoutHead)" render-mode="Static" />
</head>
<body>
    <app>
        <div class="container main">
            <component type="typeof(MainLayoutTopImages)" render-mode="Static" />
            <div class="row navmenu-row">
                <div class="col-md-12 bg-dark navmenu-col">
                    <component type="typeof(NavMenu)" render-mode="Static" />
                </div>
            </div>
            <div class="row content pt-4 pb-0 mt-0">
                <div class="col-md-12">
                    <div class="row">
                        <div class="col-md-12">
                            @*Required for GDPR.*@
                            <partial name="_CookieConsentPartial" />
                        </div>
                    </div>
                    <div class="row body-row">
                        <div class="col-md-12 body-col">
                            @RenderBody()
                        </div>
                    </div>


                </div>
            </div>
            <component type="typeof(MainLayoutFooter)" render-mode="Static" />
        </div>
    </app>

    <script src="~/Identity/lib/jquery/dist/jquery.min.js"></script>
    <script src="~/Identity/lib/bootstrap/dist/js/bootstrap.bundle.min.js"></script>
    <script src="~/Identity/js/site.js" asp-append-version="true"></script>
    @RenderSection("Scripts", required: false)
</body>
</html>
```
The `MainLayoutHead`, `MainLayoutFooter` and `NavMenu` are regular Blazor components.

**Comment by Webreaper** [comment]
I had a crack at this based on @chrissainty's article [here](https://chrissainty.com/using-blazor-components-in-an-existing-mvc-application/), trying to replace the email/password fields in my Login.cshtml file with MudBlazor `MudTextField`s, but I couldn't get it to work. Not sure if it's because they're MVC pages in an already-Blazor app and there's something I'm missing. 

That said, the point here is that all of this is a bit of a hack, and it would be much nicer if there was just a 'native' Blazor implementation of the identity pages so that it was much easier to template/style/replace the identity pages when used in a Blazor app.

**Comment by chrissainty** [comment]
This post (https://chrissainty.com/securing-your-blazor-apps-authentication-with-clientside-blazor-using-webapi-aspnet-core-identity/) is pretty old now, but it covers what you're looking to do @Webreaper. I'm not sure how much you'll need to finesse to get it working with the latest version of everything.

**Comment by Webreaper** [comment]
Thanks @chrissainty, I actually used your article as a basis to get identity into my app. It was more complex than that because I wanted a derived IdentityUser with an integer key, so I had to go through and do a load of extra stuff. It's working now, so thanks for the article!

This issue is about the fact that I'd like to customise the login, register, forgotten password, etc etc, screens to match the UX/UI styling of my app - but currently it's impossible to use controls I've built already, or third party ones like MudBlazor, because the Identity implementation is all MVC. I can potentially hack it in, but it really spoilers the easy development model of Blazor, and is jarring after writing such a large, complex and rich-UI app all in Blazor. So it feels like MSFT should embrace Blazor properly and provide a blazor identity pages implementation. 😁

**Comment by chrissainty** [comment]
The article I linked does exactly what you want to do and describes the necessary steps. I converted all the code that's used in the MVC views over to Blazor components and API controllers. It's a Blazor implementation of the Identity MVC views for signup and login.

**Comment by noseratio** [comment]
Cross-llinking this issue from my SO question, ["Can we consume a Blazor component as a Web Component within a regular non-Blazor HTML page"](https://stackoverflow.com/q/68889811/1768303). One of the other answers there mentions #35372 and @SteveSandersonMS's proposed solution to that, [`BlazorCustomElements`](https://github.com/aspnet/AspLabs/tree/main/src/BlazorCustomElements).
